File: braggbnb101_ui/src/services/LanguageService.js
import http from "../http-common"; 

class LanguageService {
  getAllLanguages(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/language/languages`, searchDTO);
  }

  get(languageId) {
    return this.getRequest(`/language/${languageId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/language?field=${matchData}`, null);
  }

  addLanguage(data) {
    return http.post("/language/addLanguage", data);
  }

  update(data) {
  	return http.post("/language/updateLanguage", data);
  }
  
  uploadImage(data,languageId) {
  	return http.postForm("/language/uploadImage/"+languageId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LanguageService();

File: src/main/java/com/braggbnb101/service/LanguageService.java
package com.braggbnb101.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb101.domain.Language;
import com.braggbnb101.dto.LanguageDTO;
import com.braggbnb101.dto.LanguageSearchDTO;
import com.braggbnb101.dto.LanguagePageDTO;
import com.braggbnb101.dto.LanguageConvertCriteriaDTO;
import com.braggbnb101.service.GenericService;
import com.braggbnb101.dto.common.RequestDTO;
import com.braggbnb101.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LanguageService extends GenericService<Language, Integer> {

	List<Language> findAll();

	ResultDTO addLanguage(LanguageDTO languageDTO, RequestDTO requestDTO);

	ResultDTO updateLanguage(LanguageDTO languageDTO, RequestDTO requestDTO);

    Page<Language> getAllLanguages(Pageable pageable);

    Page<Language> getAllLanguages(Specification<Language> spec, Pageable pageable);

	ResponseEntity<LanguagePageDTO> getLanguages(LanguageSearchDTO languageSearchDTO);
	
	List<LanguageDTO> convertLanguagesToLanguageDTOs(List<Language> languages, LanguageConvertCriteriaDTO convertCriteria);

	LanguageDTO getLanguageDTOById(Integer languageId);







}





